### ISSUE TITLE
Running Account Balance

### ISSUE THREAD

**Issue by TinShed** [firefly-iii/firefly-iii#1497]
**Description**
A running balance in Account Transactions view. 
In GnuCash each transaction shows the new/current balance of the account at the time of entry.
This would be useful for when trying to investigate any balancing or transaction errors.

Trying to import data via CSV from multiple sources is causing some duplication of transactions. A balance column would assist isolating the offending transactions.

**Solution**
An extra column "Balance" in the Transaction view of an Account (Asset, Expense & Category) which would show the account balance for each transaction.

**What are alternatives?**
None in Firefly.

**Additional context**
GnuCash has a "Balance" column in each Account. 
Most bank Statements have a running balance.

**Comment by JC5** [comment]
There is an audit report that includes this field. You can see an example on the demo site.

Due to space constraints, I can't include this in the other lists.

https://demo.firefly-iii.org/reports/audit/1/20180601/20180630

**Comment by TinShed** [comment]
Maybe give the option to select what fields are visible on the accounts view. Similar to the Option tick boxes at top of the report.
Running a report just to see the balance presents a workflow issue.

When clicking "Show all transactions without grouping them by date" within a account presents the same real estate as the audit report.

**Comment by JC5** [comment]
I'll see what I can do, there might be room to spare. I don't want to add options so I'll have to find some sensible default. But there are inconsistencies between views anyway so this might be something to work on, regardless of which fields will make the cut :)

**Comment by TinShed** [comment]
Maybe have the ability to customise the fields in the Options > Preferences page. No need for options displayed on Account view. 
(Just throwing ideas)

Thanks for your efforts!

**Comment by JC5** [comment]
You're welcome! :+1:

But those are exactly my problem. I try to make Firefly III work well "out of the box". Research suggests that [customization and personalisation](https://www.nngroup.com/articles/customization-personalization/) are difficult to do well. And just about every finance tool I know about has a billion different options. I hate that, personally.

So if anything, in Firefly II, I want it to have zero options and simply work well. I know that's impossible so instead I am very strict about the options I add.

Talking about this line:

> Trying to import data via CSV from multiple sources is causing some duplication of transactions. 

Can you give me examples of transactions that are duplicated? I'm trying to make Firefly III as smart as possible about duplications and it can already detect them _over different files_.

**Comment by TinShed** [comment]
Oh the CSV import is poor data and some user error. 

I had imported data from GnuCash fine, but thought I was missing some transactions when comparing with the actual bank account. Eg the individual transaction numbers didn't match. So I imported the raw data from my bank thinking this would just add the missing transactions.
Turns out the transaction numbers from the bank data do not match the paper statement they sent. (I'd used the paper statement in GnuCash) The paper statement had bundled multiple transactions into a single transaction, whereas the bank account showed them as 3 individual transactions. While the end balance was the same, the individual transactions were not. If I had a running balance it would have made more sense.

This is for a Home Mortgage account (using a Credit Card account type)
Some Loan fees were bundled as one figure on the statement, however the actual account had them split into 3 transactions. They did the same with the Taxes. Quite confusing!
